jetbrains-core/src/software/aws/toolkits/jetbrains/services/dynamic/DynamicResourcesUpdateManager.kt

```diff
@@ -128,29 +129,19 @@ internal class DynamicResourceUpdateManager(private val project: Project) {
 
     private fun getProgress() {
         var size = pendingMutations.size
-
+        var isValidToken = true
```

Comment:
This is set outside the loop; but only once on the inside. Which means if there are a batch of updates then the first one that fails for an invalid token, will effectively cause the rest to also be dropped.

As an aside- I really, *really* don't like `var` - even locally - I will usually look for anyway to avoid this mutability because it can lead to issues that make the code hard to reason about / bugs that are difficult to find. (This is a classic example).

What if we have the `try` block return a `Pair<ProgressEvent?, Boolean>` and then use the `Boolean` variable can drive whether we put it back on the queue again - we can also use this boolean to not-requeue items that are in a "terminal" state. So the "success" block becomes.

```kt
val (progressEvent, shouldRequeue) = try {
  val progress = client.getResourceRequestStatus { .. }
  progress.progressEvent() to !progress.progressEvent().operationStatus().isTerminal()
} catch (e: RequestTokenNotFoundException) {
..
null to false
} catch (e: Exception) {
..
null to true
}
```

Reply:
Changed to Pair